src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -177,11 +174,39 @@ public void Initialize(StorageEnvironmentOptions options)
         }
 
         private const int GuidSize = 16;
-        public long? GetChangeVectorEntryFor(DocumentsOperationContext context, Guid dbId)
+        public IEnumerable<Tuple<ArraySegment<byte>, long>> GetChangeVector(DocumentsOperationContext context)
```

Comment:
Complex, and exposes internal state, why?
You are better off defining a struct { Guid, long} and then passing an array of that.
Simpler, and cheaper.